pkg/replica/diff_disk.go

```diff
@@ -304,10 +308,15 @@ func (d *diffDisk) UnmapAt(unmappableDisks []string, length uint32, offset int64
 	if endSectorOffset != 0 {
 		length -= uint32(endSectorOffset)
 	}
-	if length <= 0 {
+	if length == 0 {
 		return 0, nil
 	}
 
+	// final check
+	if origLength < length {
```

Comment:
Is it possible?

Reply:
No, I thought it was impossible after this patch.
I would like to have this because I am afraid I missed some cases.
We could avoid the unexpected punch hole length and get the error.

WDYT?

Reply:
I'm fine with it. Can you add a comment for the check to avoid any confusion?

Reply:
Sure, updated the comment!

Reply:
the unexpected punch hole length? I was thinking it handles `length` being overflowed to a large number

Reply:
The length being overflowed to a large number means it might be a large number for the punch hole length.
That's why I say the unexpected punch hole length.

This checks here because I want to create some punch hole protection.
The final length is bigger than the original length, which means the extra sector will be punch hole. It caused data corruption.

I update the comment again. Hope that might be more clear.
Thanks!